worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/ContractAdapterImpl.java

```diff
@@ -40,68 +42,59 @@
     private final BFDClient bfdClient;
     private final ContractRepository contractRepo;
     private final BeneficiaryService beneficiaryService;
+    private final PropertiesService propertiesService;
 
 
     @Override
     public GetPatientsByContractResponse getPatients(final String contractNumber, final int currentMonth) {
 
         var patientDTOs = new ArrayList<PatientDTO>();
 
+        final boolean cachingOn = isContractToBeneCachingOn();
+
         var contract = contractRepo.findContractByContractNumber(contractNumber).get();
-        var contractId = contract.getId();
 
         for (var month = 1; month <= currentMonth; month++) {
-
-            Set<String> bfdPatientsIds = beneficiaryService.findPatientIdsInDb(contractId, month);
-            if (bfdPatientsIds.isEmpty()) {
-                // patient ids were not found in local DB given the contractId and currentMonth
-                // call BFD to fetch the data
-
-                bfdPatientsIds = getPatientIdsForMonth(contractNumber, month);
-
-                //if number of benes for this month exceeds cachingThreshold, cache it
-                var beneficiaryCount = bfdPatientsIds.size();
-                if (beneficiaryCount > cachingThreshold) {
-                    beneficiaryService.storeBeneficiaries(contract.getId(), bfdPatientsIds, month);
-                }
-            }
+            var bfdPatientsIds = getPatientsForMonth(contractNumber, contract, month, cachingOn);
 
             var monthDateRange = toDateRange(month);
 
             for (String bfdPatientId : bfdPatientsIds) {
+                buildPatientDTOs(patientDTOs, monthDateRange, bfdPatientId);
+            }
+        }
+
+        return toGetPatientsByContractResponse(contractNumber, patientDTOs);
+    }
 
-                var optPatient = findPatient(patientDTOs, bfdPatientId);
-                if (optPatient.isPresent()) {
-                    // patient id was already active on this contract in previous month(s)
-                    // So just add this month to the patient's dateRangesUnderContract
+    private boolean isContractToBeneCachingOn() {
+        return propertiesService.isToggleOn("ContractToBeneCachingOn");
+    }
```

Comment:
The tests for properties are in PropertiesServiceTest, you can add this one to there.

Reply:
done